docs/okeps/okep-5233-pre-assigned-pod-net-id.md

```diff
@@ -0,0 +1,455 @@
+# OKEP-5233: Pre-assigned network configuration for primary user defined networks workloads
+
+* Issue: [#5233](https://github.com/ovn-org/ovn-kubernetes/issues/5233)
+
+## Problem Statement
+
+Migrating legacy workloads with pre-assigned network configurations (IP, MAC, default gateway)
+to OVN-Kubernetes is currently not possible. There is a need to import these workloads, preserving 
+their network configuration, while also enabling non-NATed traffic to better integrate with
+existing infrastructures.
+
+## Goals
+
+- Enable pods on primary Layer2 User Defined Network (UDN) to use a pre-defined static network
```

Comment:
Do we know if OVNK support pod migration if we want to see how this behavior works on pods in liu of VMs? AFAIK, pods are ephermeral on OVNK. I am wondering If a pod is accidentally deleted and recreated, would this feature ensure its network configuration like IP, MAC persists in L2 scenario?

Reply:
Probably also would be interesting to find out on how to handle scenarios where the static IP range overlaps with dynamically assigned IPs in the same UDN?

Reply:
> Do we know if OVNK support pod migration if we want to see how this behavior works on pods in liu of VMs? AFAIK, pods are ephermeral on OVNK. I am wondering If a pod is accidentally deleted and recreated, would this feature ensure its network configuration like IP, MAC persists in L2 scenario?

this feature is currently only for pods that hold VMs. It just does not work for a "regular" pod .

I agree it could make sense to a stateful set pod though, but the OVN-K implementation would need to change to accommodate that .

FWIW, that's a new feature, and nobody's requesting it.

But - on paper - it looks promising. I don't see how the same concept / solution would not work.